### PR TITLE
Initialize multiple language and start translations

### PR DESCRIPTION
**Pull request by anddimario**
### Issue
https://github.com/mrvautin/expressCart/issues/87

### Todo
- linting
- review

### CODE REVIEW

**Comment by mrvautin** [comment]
Looks ok so far. There is a heap of linting. Are you using the eslint file in the project?

**Comment by anddimario** [comment]
Yes, i'm using it, i'll investigate about that

**Comment by anddimario** [comment]
I think that i've complete the translation (en and it), about linting, it seems that are the same master repository results (i've not edit lots of js files).
If you agree, i'll set the pr for the review.

**Comment by mrvautin** [comment]
Look good mate. Seems there is still some whitespace linting happening in the `getStatusColor` function of the `app.js` file. 

Maybe switch to review and I can highlight anything I see.

**Comment by anddimario** [comment]
It seems that there're some conflict, i'll wait your input to know how you would proceed :)

**Comment by mrvautin** [comment]
Could you try rebasing back onto master? Most should not be conflicts.

**Comment by anddimario** [comment]
I've merged upstream into my fork and solved conflict, could it be good for you?

File: app.js [hunk 1]
```diff
@@ -272,6 +293,11 @@ app.use((req, res, next) => {
     next();
 });
 
+app.use((req, res, next) => {
+    res.locals.languages = i18n.getLocales();
```
Comment (mrvautin) [inline app.js]: Can this be moved to line 271 with the `req.handlebars = handlebars;` line?

File: config/settings.json [hunk 2]
```diff
@@ -16,13 +16,15 @@
     "productsPerPage": 6,
     "menuTitle": "Menu",
     "menuLocation": "side",
-    "footerHtml": "&lt;h4 class=&quot;text-center&quot;&gt;Powered by expressCart&lt;/h4&gt;",
```
Comment (mrvautin) [inline config/settings.json]: Can you revert all changes in the `settings.json file` except the new additions?

File: package.json [hunk 3]
```diff
@@ -29,12 +30,14 @@
...
     "helmet": "^3.21.2",
     "html-entities": "^1.2.0",
+    "i18n": "^0.8.4",
     "jsonschema": "^1.2.4",
     "lodash": "^4.17.15",
     "lunr": "^2.3.8",
     "mime-type": "^3.0.7",
     "mkdirp": "^0.5.1",
     "moment": "^2.24.0",
+    "moment-recur": "^1.0.7",
```
Comment (mrvautin) [inline package.json]: This appears unused. Can it be removed?

File: views/customers.hbs [hunk 4]
```diff
@@ -11,29 +11,29 @@
                 <a href="/admin/customers" class="hidden-xs btn btn-warning btn-lg"><i class="fa fa-times" aria-hidden="true"></i></a>
             </span>
         </div>
-        <p class="text-warning top-pad-10">Customers can be filtered by: email, name or phone numnber</p>
```
Comment (mrvautin) [inline views/customers.hbs:14]: `numnber` typo. Can you change to `number`. Also need to update translation file.

File: views/themes/Cloth/payment_complete.hbs [hunk 5]
```diff
@@ -2,18 +2,18 @@
     <div class="row">
         <div class="text-center col-md-10 col-md-offset-1">
             {{#ifCond result.orderStatus '==' 'Paid'}}
-                <h2 class="text-success">Your payment has been successfully processed</h2>
+                <h2 class="text-success">{{ @root.__ "Your payment has been successfully processed" }}</h2>
             {{else}}
-                <h2 class="text-danger">Your payment has failed. Please try again or contact us.</h2>
```
Comment (mrvautin) [inline views/themes/Cloth/payment_complete.hbs:7]: Looks like a duplicate. Should have the failed message.

File: views/themes/Material/payment_complete.hbs [hunk 6]
```diff
@@ -2,18 +2,18 @@
     <div class="row">
         <div class="text-center col m10 offset-m1">
             {{#ifCond result.orderStatus '==' 'Paid'}}
-                <h2 class="text-success">Your payment has been successfully processed</h2>
+                <h2 class="text-success">{{ @root.__ "Your payment has been successfully processed" }}</h2>
             {{else}}
-                <h2 class="text-danger">Your payment has failed. Please try again or contact us.</h2>
+                <h2 class="text-danger">{{ @root.__ "Your payment has been successfully processed" }}</h2>
```
Comment (mrvautin) [inline views/themes/Material/payment_complete.hbs]: This should have the failed message.

File: views/themes/Mono/payment_complete.hbs [hunk 7]
```diff
@@ -2,18 +2,18 @@
     <div class="row">
         <div class="text-center col m10 offset-m1">
             {{#ifCond result.orderStatus '==' 'Paid'}}
-                <h2 class="text-success">Your payment has been successfully processed</h2>
+                <h2 class="text-success">{{ @root.__ "Your payment has been successfully processed" }}</h2>
             {{else}}
-                <h2 class="text-danger">Your payment has failed. Please try again or contact us.</h2>
+                <h2 class="text-danger">{{ @root.__ "Your payment has been successfully processed" }}</h2>
```
Comment (mrvautin) [inline views/themes/Mono/payment_complete.hbs]: This should have the failed message.

File: app.js [hunk 8]
```diff
@@ -123,22 +141,22 @@ handlebars = handlebars.create({
         },
         getStatusColor: (status) => {
             switch(status){
-            case'Paid':
```
Comment (mrvautin) [inline app.js:126]: Lots of whitespace changes. Is your eslint setup correctly?

**Review by mrvautin — CHANGES_REQUESTED** [review]
Looks good mate. Just a few comments. I appreciate your contributions.

Seems like your eslint isn't working as expected as there are a lot of whitespace changes which don't show on `master`.

**Comment by anddimario** [comment]
i think that i've solved errors

**Comment by mrvautin** [comment]
Thanks. I cannot merge as there are more conflicts. Can you do one last rebase onto master and I will merge.

**Comment by anddimario** [comment]
it seems that i can fetch nothing from upstream, and here github doesn't show conflicts, are you sure that you don't refer to a specific branch?

**Review by mrvautin — APPROVED** [review]

**Comment by mrvautin** [comment]
It seems Github has changed. I've merged your changes now. Thanks.

**Pull request merged**